src/admin-portal/create-campaign/campaign-testimonial-view.js

```diff
@@ -1,66 +1,61 @@
-import React, { useReducer, useState } from "react";
-import { Col, Container, Form, Modal, Row, Spinner } from "react-bootstrap";
+import React, { useReducer, useRef, useState } from "react";
+import { Col, Container, Button as BTN, Modal, Row } from "react-bootstrap";
 import TestimonialCard from "../../components/admin-components/TestimonialCard";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faAd, faAdd } from "@fortawesome/free-solid-svg-icons";
-import { fetchAllTechnologyTestimonials } from "src/requests/technology-requests";
+import { fetchAllCampaignTestimonials } from "src/requests/technology-requests";
 import useSWR from "swr";
 import Testimonials from "../../components/admin-components/Testimonials";
 import { Testimonials as TestPortal } from "./create-technology/Testimonials";
+import Loading from "src/components/pieces/Loading";
 
 const opts = ["All", "Featured"];
 const allOpts = ["On Campaign", "From Communities"];
 
-export function CampaignTestimonialsView({ campaignDetails, }) {
+export function CampaignTestimonialsView({ campaignDetails }) {
   const [show, setShow] = useState({
     tabOne: opts[0],
-    tabTwo: allOpts[0],
-    showFormFor: "campaign",
+    tabTwo: "On Campaign",
+    showFormFor: "",
   });
 
   const [openModal, setOpenModal] = useState(false);
 
+  const formRef = useRef();
+
+  const startOfPage = useRef(null);
+
   let {
     data: payloadTestimonials,
     isValidating,
     isLoading,
     error,
-  } = useSWR(
-    "testimonials.list",
-    async () => await fetchAllTechnologyTestimonials(campaignDetails?.id),
-    {
-      shouldRetryOnError: true,
-      errorRetryCount: 3,
-      errorRetryInterval: 3000,
-    },
-  );
-
-  const technologies = campaignDetails?.technologies;
-  const testimonials = [].concat(...technologies?.map((tech) => tech?.testimonials));
+  } = useSWR("campaigns.testimonials.list", async () => await fetchAllCampaignTestimonials(campaignDetails?.id), {
+    shouldRetryOnError: true,
+    errorRetryCount: 3,
+    errorRetryInterval: 3000,
+  });
 
-  const [data, setData] = useState(testimonials);
+  const testimonials = payloadTestimonials || [];
 
-  const portalTestimonials = data?.filter((test) => {
+  const portalTestimonials = testimonials?.filter((test) => {
```

Comment:
I really like the descriptive names used to replace the more generic variable names.
Great job!